cpp/src/join/full_join.cuh

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "join_common_utils.hpp"
+#include <cudf/copying.hpp>
+
+namespace cudf {
+
+namespace detail {
+
+template <typename size_type>
+struct ValidRange {
```

Comment:
Camel case in libcudf is only used for some template parameters and for test fixture names.
```suggestion
struct valid_range {
```